plugins/woocommerce/client/blocks/assets/js/blocks/product-gallery/style.scss

```diff
@@ -164,96 +174,83 @@ $dialog-padding: 20px;
 		object-fit: contain;
 	}
 
-	&__thumbnail__overlay {
-		container-type: inline-size;
+	&::before,
+	&::after {
+		content: "";
 		position: absolute;
-		top: 0;
-		left: 0;
-		width: 100%;
-		height: 100%;
-		display: none;
-		flex-direction: column;
-		align-items: center;
-		justify-content: center;
-		cursor: pointer;
-		color: #fff;
-		background-color: rgba(0, 0, 0, 0.4);
-		line-height: 1.5;
+		opacity: 0;
+		transition: opacity 0.2s linear;
+		pointer-events: none;
+	}
 
-		&[visible="true"] {
-			display: flex;
+	&.wc-block-product-gallery-thumbnails--overflow {
+		&-top::before,
+		&-bottom::after {
+			left: 0;
+			height: $thumbnails-gradient-size;
+			width: 100%;
+			opacity: 1;
 		}
 
-		.wc-block-product-gallery-thumbnails__thumbnail__remaining-thumbnails-count {
-			font-size: 1.2rem;
-			font-weight: 700;
+		&-top::before {
+			top: 0;
+			@include gradient(bottom);
 		}
 
-		.wc-block-product-gallery-thumbnails__thumbnail__view-all {
-			text-decoration: underline;
+		&-bottom::after {
+			bottom: 0;
+			@include gradient(top);
 		}
 
-		@container (width < 70px) {
-			.wc-block-product-gallery-thumbnails__thumbnail__view-all {
-				display: none;
-			}
-
-			.wc-block-product-gallery-thumbnails__thumbnail__remaining-thumbnails-count {
-				font-size: 1rem;
-			}
-
-			.wc-block-product-gallery-thumbnails__thumbnail__view-all {
-				font-size: 0.6rem;
-			}
+		&-right::after,
+		&-left::before {
+			top: 0;
+			height: 100%;
+			width: $thumbnails-gradient-size;
+			opacity: 1;
 		}
 
-		@container (width > 70px) {
-			.wc-block-product-gallery-thumbnails__thumbnail__remaining-thumbnails-count {
-				font-size: 1.2rem;
-			}
-
-			.wc-block-product-gallery-thumbnails__thumbnail__view-all {
-				font-size: 0.8rem;
-			}
+		&-left::before {
+			left: 0;
+			@include gradient(right);
 		}
 
-		@container (width > 100px) {
-			.wc-block-product-gallery-thumbnails__thumbnail__remaining-thumbnails-count {
-				font-size: 1.4rem;
-			}
-
-			.wc-block-product-gallery-thumbnails__thumbnail__view-all {
-				font-size: 1rem;
-			}
+		&-right::after {
+			right: 0;
+			@include gradient(left);
 		}
 	}
 }
 
-// Automattically react to the Group container settings.
+// Percentage width
 .is-nowrap .wc-block-product-gallery-thumbnails,
 .is-horizontal.is-nowrap .wc-block-product-gallery-thumbnails {
-	flex-direction: column;
+	&__scrollable {
+		flex-direction: column;
+	}
 
-	@for $i from 3 through 8 {
-		&.wc-block-product-gallery-thumbnails--number-of-thumbnails-#{$i} {
+	@for $i from 10 through 50 {
```

Comment:
I would add a comment that `10` and `50` are related to the `min`, `max` block values.